### DISCUSSION TITLE
Multiple $refs

### DISCUSSION THREAD

**Discussion by yudyananda** [alpinejs/alpine#1778] [1. Help] [answered]
is it possible to have multiple `$refs`? 

```html
<button @click="$refs.foo.innerText='foo' $refs.bar.innerText='bar'">A button</button>

<span x-ref="foo"></span>
<span x-ref="bar"></span>
```

**Comment by danddanddand** [comment, marked as answer]
Yes. From your example cleaned up a bit:

https://codepen.io/danddanddand/pen/zYwBRdG

**Reply by yudyananda** [reply]
Awesome!  does semicolon is a must? i'm curious because the ref will do nothing if i remove it

**Reply by danddanddand** [reply]
You could use newlines to separate two statements or semicolon like regular JS
